File: api/src/app/states/dto/update-state.dto.ts
import { PartialType } from '@nestjs/mapped-types';
import { CreateStateDto } from '@app/states/dto/create-state.dto';

export class UpdateStateDto extends PartialType(CreateStateDto) {
  is_active: boolean;
  is_backlog: boolean;
  is_canceled: boolean;
}

File: api/src/main/java/fr/tempest/tempestboard/state/UpdateStateDto.java
package fr.tempest.tempestboard.state;

public class UpdateStateDto extends CreateStateDto {
    private Boolean isActive;
    private Boolean isBacklog;
    private Boolean isCanceled;

    public Boolean getActive() {
        return isActive;
    }

    public void setActive(Boolean active) {
        isActive = active;
    }

    public Boolean getBacklog() {
        return isBacklog;
    }

    public void setBacklog(Boolean backlog) {
        isBacklog = backlog;
    }

    public Boolean getCanceled() {
        return isCanceled;
    }

    public void setCanceled(Boolean canceled) {
        isCanceled = canceled;
    }
}
